djangocms_navigation/admin.py

```diff
@@ -467,6 +535,79 @@ def changelist_view(self, request, menu_content_id=None, extra_context=None):
 
         return super().changelist_view(request, extra_context)
 
+    def delete_view(
+            self, request, object_id, menu_content_id=None, form_url="", extra_context=None
+    ):
+        extra_context = extra_context or {}
+        if menu_content_id:
+            request.menu_content_id = menu_content_id
+            list_url = reverse_admin_name(
+                self.model,
+                'list',
+                kwargs={"menu_content_id": menu_content_id},
+            )
+            extra_context["list_url"] = list_url
+            if self._versioning_enabled:
+                menu_content = get_object_or_404(
+                    self.menu_content_model._base_manager, id=menu_content_id
+                )
+                delete_perm = self.has_delete_permission(request, menu_content)
+                if not delete_perm:
+                    messages.error(request, LOCK_MESSAGE)
```

Comment:
Seems to be a gap in the test coverage as this lock message is not checked in deletion, see comments in the tests below.